src/components/app.jsx

```diff
@@ -33,7 +36,21 @@ class App extends React.Component {
     }
   }
 
+  saveCards() {
+    const myCards = JSON.stringify(this.state.cards);
+    localStorage.setItem("flash-cards", myCards);
+  }
+
+  addCard(card) {
+    const myCards = this.state.cards.map(element => ({...element}));
+    myCards.push(card);
+    this.setState({
+      cards: myCards
+    }, this.saveCards());
```

Comment:
We shouldn't be calling save cards here, it will happen before the state is updated. We want to pass `this.saveCards` as a callback function.

Reply:
Ok, I think I fixed it to be a callback function

![image](https://user-images.githubusercontent.com/66789858/92813821-9ab66280-f387-11ea-868c-e51533ca6368.png)